src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java

```diff
@@ -374,20 +376,50 @@ public Point point(final long identifier)
     }
 
     /**
-     * In case of an Atlas shard with neighbors containing data that might intersect it (example is
-     * a way from a neighboring shard intersecting the initial shard, but without any shapepoints in
-     * the initial shard) this method allows dual loading of neighboring shards. Once to load the
-     * neighboring shards if any, and to discover such features if any, then to load the other
-     * shards those features might intersect.
+     * Do a preemptive load of the {@link DynamicAtlas} as far as the {@link DynamicAtlasPolicy}
+     * allows.
+     * <p>
+     * In some very specific case, where the {@link DynamicAtlasPolicy} allows expansion only if new
+     * shards intersect at least one feature that crosses the initial set of shards, it is possible
+     * that expanding only one time misses out some shard candidates. This happens when some feature
+     * intersects the initial shards but does not have any shape point inside any initial shard.
+     * This way, the initial shards do not contain that feature even though they intersect it. That
+     * feature is discovered as we load the neighboring shards which contain that feature. If that
+     * said feature also intersects a third neighboring shard, then that third neighboring shard
+     * becomes eligible for expansion, as that specific feature crosses it and the initial shards.
+     * To work around that case, the preemptive load will do a multi-staged loading.
      */
     public void preemptiveLoad()
     {
-        Set<Shard> currentShards = null;
+        // Loop through the entities to find potential shards to add
+        this.entities();
+        // Load all the shards into a multiAtlas
+        this.buildUnderlyingMultiAtlas();
+        // Record the current list of shards
+        Set<Shard> currentShards = new HashSet<>(this.loadedShards.keySet());
+        // Loop through the entities again to find potential shards to add. This can still happen if
+        // a way intersects the initial shard without shapepoints inside the initial shards, and was
```

Comment:
stupid question: when would a way intersect a shard without shape points inside the shard?  Would that be when the way ends exactly on the boundary of the shard?  And the case of a third shard, would that be a way ends exactly on the boundary of where multiple shards meet?

Reply:
Here is the fake data I created for the test in that PR:
<img width="1075" alt="screen shot 2017-12-20 at 9 48 35 am" src="https://user-images.githubusercontent.com/1944860/34220889-f909398a-e56a-11e7-94b9-1a1e5c284241.png">
You see that the lower left shard is intersected by the red way, but none of its shapepoints are inside.

Reply:
And if created from an osmosis PBF, the lower left shard will not contain the red way. It will be discovered only when expanding to the upper left or lower right shards. Then that uncovers a new way that intersects the initial shard, which in turn makes the upper right shard eligible for loading on the second pass, when it was not initially.

Reply:
Oh I see! Thanks for the explanation